### DISCUSSION TITLE
Get user to verify and set password on account creation

### DISCUSSION THREAD

**Discussion by DigNZ** [pocketbase/pocketbase#3552] [Q&A] [answered]
I have a web interface where managers can create users. What is the best approach to force these users to verify AND set a password so the managers are not setting passwords?

I feel like its something in here:
`app.OnRecordAfterCreateRequest("briefs").Add(...`
and I could send maybe send both the verification email and the reset password email from there but I think there is a better way.

**Comment by ganigeorgiev** [comment, marked as answer]
In this case I'd send only a password reset email.

You can automatically change the user verified state in the [`OnRecordBeforeConfirmPasswordResetRequest` hook](https://pocketbase.io/docs/go-event-hooks/#onrecordbeforeconfirmpasswordresetrequest) by doing something like `e.Record.SetVerified(true)`.

If you want a custom email template, you can also attach to [`OnMailerBeforeRecordResetPasswordSend` hook](https://pocketbase.io/docs/go-event-hooks/#onmailerbeforerecordresetpasswordsend) and for example add a condition to change the `e.Message.HTML` if the user record is not verified yet.

**Reply by ganigeorgiev** [reply]
_I've added a task in the [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1) to consider implementing a predefined invitation flow sometime during the auth refactoring as it was requested several times in the past._

**Reply by DigNZ** [reply]
Thanks as always. I'll keep an eye on it.
